lib/vector-core/src/event/util/log/all_fields.rs

```diff
@@ -10,14 +10,14 @@ static IS_VALID_PATH_SEGMENT: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[a-zA-Z0-
 
 /// Iterates over all paths in form `a.b[0].c[1]` in alphabetical order
 /// and their corresponding values.
-pub fn all_fields(fields: &ObjectMap) -> FieldsIter {
-    FieldsIter::new(fields)
+pub fn all_fields(fields: &ObjectMap, quote_periods: bool) -> FieldsIter {
```

Comment:
With `quote_periods = true`, the current behavior of the iterator is preserved, as demonstrated by the tests. It only changes the behavior when that parameter is `false`. I don't want the old behavior where periods are escaped neither, as that too caused problems in the `new_relic` sink.

Reply:
Plain `bool` parameters in utils can be an anti-pattern. Also, eventually all these iterator should return `OwnedTargetPath`s. Are the paths returned by this new iterator parsable with `parse_target_path`?

Reply:
I have split up the `all_fields` function. No, the unquoted paths are no longer parsable. Technically, I don't think the current paths are all parsable neither if they contain quotes since those are not escaped.